.gitlab-ci.yml

```diff
@@ -0,0 +1,215 @@
+
+stages:
+  - buildenv
+  - build
+  - test
+
+.fedora-rawhide: &fedora-rawhide-anchor
+  tags:
+    - docker
+  image: fedora:rawhide
+
+.buildenv: &buildenv-anchor
+  stage: buildenv
+  variables:
+    GIT_STRATEGY: none
+
+.compilation: &compilation-anchor
+  only:
+   - pushes
+  # only since GitLab 11.4
+  #  changes:
+  #    - "configure.ac"
+  #    - "**/*.[ch]"
+  #    - "**/*.xml"
+  #    - "**/*.xsl"
+  #    - "**/Makefile.am"
+  #    - ".gitlab-ci.yml"
+
+buildenv:fedora-rawhide:info:
+  # primarily to gather some info about the image itself/container settings
+  <<: *buildenv-anchor  #extends: .buildenv
+  <<: *fedora-rawhide-anchor  #extends: .fedora-rawhide
+  script:
+   - cat /proc/meminfo | grep Free
+   - df -H
+   - lsblk
+   - nproc
+   - ulimit -a
+   - dnf --version
+
+buildenv:fedora-rawhide:rpms:
+  # primarily to download RPMs, but alas, "download" plugin not baked into image
+  <<: *buildenv-anchor  #extends: .buildenv
+  <<: *fedora-rawhide-anchor  #extends: .fedora-rawhide
+  cache:
+    key: "fedora-rawhide-dnf"
+    paths:
+      - ".cache/fedora-rawhide/dnf/rawhide-*"
+  script:
+    - mkdir -p .cache/fedora-rawhide/dnf
+    - cd .cache/fedora-rawhide
+    - mv dnf/rawhide-* /var/cache/dnf 2>/dev/null || true
+    - dnf-custom() { timeout 192 dnf -y --noplugins --nodocs
+                                 --setopt=tsflags=test
+                                 --setopt=install_weak_deps=0
+                                 --setopt=keepcache=1
+                                 --setopt=autocheck_running_kernel=0
+                                 --disablerepo='*' --enablerepo=rawhide "$@"; };
+      dnf-custom install
+        autoconf automake pkgconf-pkg-config libtool libtool-ltdl-devel
+        make findutils
+        gcc clang compiler-rt
+        libuuid-devel libxslt-devel bzip2-devel
+        'pkgconfig(glib-2.0) >= 2.16.0' 'pkgconfig(libxml-2.0)'
+        'pkgconfig(libqb) >= 0.13' 'pkgconfig(libcpg)'
+        libxml libxslt
+        psmisc which
+        bzip2-libs glib2
+        gcovr
+    - mv /var/cache/dnf/rawhide-* dnf
+
+maint:fedora-rawhide:
+  <<: *fedora-rawhide-anchor  #extends: .fedora-rawhide
+  stage: build
+  cache:
+    key: "fedora-rawhide-dnf"
+    policy: pull
+    paths:
+      - ".cache/fedora-rawhide/dnf/rawhide-*"
+  before_script:
+    - mkdir -p .cache/fedora-rawhide/dnf
+    - pushd .cache/fedora-rawhide
+    - mv dnf/rawhide-* /var/cache/dnf 2>/dev/null || true
+    - dnf-custom() { timeout 96 dnf -y --cacheonly --noplugins --nodocs
+                                --setopt=install_weak_deps=0
+                                --setopt=autocheck_running_kernel=0
+                                --disablerepo='*' --enablerepo=rawhide "$@"; };
+      dnf-custom install
+        findutils make libxml libxslt
+    - popd
+    - ( cd xml;
+        { cat Makefile.am; printf 'hack_rng\x3a %s' '${RNG_generated}'; }
+          | make -f- top_srcdir=$(pwd)/.. top_builddir=$(pwd)/.. hack_rng )
```

Comment:
It would be fine to add a real target for this in this PR

Reply:
It would be along the lines of moving something into static `xml/GNUMakefile`.
Since this just mirrors what's in `.travis.yml` already, there's a scope creep
of the PR, but I can keep it here if you wish.

Reply:
Hmm, let's leave it for now. Might be worthwhile eventually though.